cloudevents/pydantic/event.py

```diff
@@ -0,0 +1,240 @@
+#  Copyright 2018-Present The CloudEvents Authors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import datetime
+import typing
+
+import pydantic
+
+from cloudevents.pydantic import event_attribute_types
+from cloudevents.pydantic.json_format import ce_json_dumps, ce_json_loads
+from cloudevents.sdk.event import defaults
+
+
+class CloudEvent(pydantic.BaseModel):
```

Comment:
I'd overall say that I'd rather see Pydantic model as smth which extends the base `CloudEvent` class. This way one can use the same conversion APIs.

Reply:
I'm not sure this is possible, will try.

Reply:
So there's no way to extend the base model from Pydantic and our CloudEvent together?

Reply:
it creates complexities with the `_attributes` data member and keep the consistency between the pydantic class attributes and the `_attributes` data member.

I think it is much easier to convert between the two representations

Reply:
Actually come to think about it. I can create an abstract CloudEvent interface which MAY be inherited by the http event and the pydantic event. 

This will solve all of these issues

Reply:
I also think that we should look towards a solution where the Pydantic model inherits from the base CloudEvent model. Pydantic adds some delays to the initialization of a class and I'd rather not pay that. 

We can also make this an extension class where the Pydantic class is totally separate from the CloudEvent class. If someone desires the additional validation, they can use that class instead of the raw CloudEvent class.

Reply:
@rednafi @xSAVIKx take a look at my recent changes. Now the `pydantic.CloudEvent` and `http.CloudEvent` inherit from `abstract.CloudEvent`